js/app.js

```diff
@@ -48,10 +50,18 @@ $(document).ready(function() {
 
   // Input binding
   $searchInput
-  .on('keyup', function() {
-    var query = $(this).val();
-    toggleIconEmptyInput(query);
-    algoliaHelper.setQuery(query).search();
+  .on('keyup paste cut change', function() {
+    var $el = $(this);
+
+    // We use setTimeout to defer the callback to the bottom of the stack to
+    // catch the value after the paste, cut, or delete via mouse
+    setTimeout(function(){
+      var query = $el.val();
```

Comment:
Seems like the `input` event is a magical one that can help us but I cannot find a jquery reference to it :)